spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+                                $"SingerId : {reader["SingerId"]}"
+                                + $" AlbumId : {reader["AlbumId"]}"
+                                + $" AlbumTitle : {reader["AlbumTitle"]}");
+                        }
+                    }
+                }
+                // Commit the transaction to Spanner table.
+                scope.Complete();
+                // Yield Task thread back to the current context.
+                await Task.Yield();
+                return;
+                // [END read_only_transaction]
+            }
+        }
+
+        public static async Task ReadSampleDataAsync(
+    string projectId, string instanceId, string databaseId)
+        {
+            // [START query_data]
+            // [START read_data]
+            Console.WriteLine("Querying data...");
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+            // Create connection to spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                Console.WriteLine("Connected...");
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT SingerId, AlbumId, AlbumTitle FROM Albums");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"SingerId = {reader["SingerId"]} "
+                        + $"AlbumId = {reader["AlbumId"]} "
+                        + $"AlbumTitle = {reader["AlbumTitle"]}");
+                    }
+                }
+            }
+            return;
+            // [END read_data]
+            // [END query_data]
+        }
+
+        public static async Task WriteDataToNewColumnAsync(
+    string projectId, string instanceId, string databaseId)
+        {
+            // [START update_data]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+            // Create connection to spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateUpdateCommand("Albums",
+                    new SpannerParameterCollection {
+                        {"SingerId", SpannerDbType.Int64},
+                        {"AlbumId", SpannerDbType.Int64},
+                        {"MarketingBudget", SpannerDbType.Int64},
+                    });
+                var cmdLookup =
+                    connection.CreateSelectCommand("SELECT * FROM Albums");
+                using (var reader = await cmdLookup.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        if (reader.GetInt64(0) == 1 && reader.GetInt64(1) == 1)

```

Comment:
Actually why do you need to lookup this row? You can assume it is there. It will keep the sample simpler.